src/timepicker/timepicker.ts

```diff
@@ -40,68 +40,70 @@ const NGB_TIMEPICKER_VALUE_ACCESSOR = {
     }
   `],
   template: `
-    <table>
-      <tr>
-        <td class="text-xs-center">
-          <button class="btn-link" (click)="changeHour(hourStep)"><span class="chevron"></span></button>
-        </td>
-        <td>&nbsp;</td>
-        <td class="text-xs-center">
-          <button class="btn-link" (click)="changeMinute(minuteStep)"><span class="chevron"></span></button>
-        </td>
-        <template [ngIf]="seconds">
-          <td>&nbsp;</td>
+    <fieldset [disabled]="disabled">
+      <table>
+        <tr>
           <td class="text-xs-center">
-            <button class="btn-link" (click)="changeSecond(secondStep)"><span class="chevron"></span></button>
+            <button class="btn-link" (click)="changeHour(hourStep)"><span class="chevron"></span></button>
           </td>
-        </template>
-        <template [ngIf]="meridian">
-          <td>&nbsp;</td>
           <td>&nbsp;</td>
-        </template>
-      </tr>
-      <tr>
-        <td>
-          <input type="text" class="form-control" maxlength="2" size="2" 
-            [ngModel]="formatHour(model?.hour)" (ngModelChange)="updateHour($event)">
-        </td>
-        <td>&nbsp;:&nbsp;</td>
-        <td>
-          <input type="text" class="form-control" maxlength="2" size="2"
-            [ngModel]="formatMinSec(model?.minute)" (ngModelChange)="updateMinute($event)">
-        </td>
-        <template [ngIf]="seconds">
+          <td class="text-xs-center">
+            <button class="btn-link" (click)="changeMinute(minuteStep)"><span class="chevron"></span></button>
+          </td>
+          <template [ngIf]="seconds">
+            <td>&nbsp;</td>
+            <td class="text-xs-center">
+              <button class="btn-link" (click)="changeSecond(secondStep)"><span class="chevron"></span></button>
+            </td>
+          </template>
+          <template [ngIf]="meridian">
+            <td>&nbsp;</td>
+            <td>&nbsp;</td>
+          </template>
+        </tr>
+        <tr>
+          <td>
+            <input type="text" class="form-control" maxlength="2" size="2" [readonly]="readonly"
```

Comment:
I don't see `readonly` field defined anywhere in the component. Did you mean to use `disabled`?